app/cancel/get.controller.js

```diff
@@ -5,12 +5,15 @@ const connectorClient = require('../../common/clients/connector-client')
 const {getSessionVariable} = require('../../common/config/cookies')
 
 module.exports = (req, res) => {
-  const paymentRequestExternalId = req.params.paymentRequestExternalId
-  const paymentRequest = getSessionVariable(req, paymentRequestExternalId).paymentRequest
+  const paymentRequest = res.locals.paymentRequest
+  const paymentRequestExternalId = paymentRequest.externalId
+  const gatewayAccountExternalId = paymentRequest.gatewayAccountExternalId
+  const returnUrl = paymentRequest.returnUrl
```

Comment:
This looks great but performance wise it’s maybe not worth setting a const for something that is only used once. @brid is this true? Maybe could just set `paymentRequest` and the rest come from that one?

Reply:
Yes, agreed. I have changed that.

Reply:
performance wise, you can argue that 'const' leaves things up to the garbage collector to dispose of references against which in the case of things holding onto functions/closures or other complexity can degrade performance compared to using a 'let' and explicitly setting this to undefined.

Obviously this depends on how hard you are hitting a certain piece of code, in what circumstances, on what hardware etc etc etc.